provider/blob/src/blob_schema.rs

```diff
@@ -24,16 +24,20 @@ pub(crate) struct BlobSchemaV1<'data> {
     /// Weak invariant: there is at least one value for every integer in 0..self.buffers.len()
     #[serde(borrow)]
     pub keys: ZeroMap2dBorrowed<'data, DataKeyHash, Index32U8, usize>,
-    /// Vector of buffers
+    /// Indices into the buffer slice
     #[serde(borrow)]
-    pub buffers: &'data VarZeroSlice<[u8], Index32>,
+    pub indices: &'data FlexZeroSlice,
+    /// Buffer slice
+    #[serde(borrow)]
+    pub buffer: &'data [u8],
```

Comment:
Note: I can probably get a further boost by using serde_bytes::Bytes or a similar type (likely an internal one) at this spot, so that we use the deserialize_bytes code path instead of deserialize_seq.